plugins/hashsum_download/server/__init__.py

```diff
@@ -118,7 +148,50 @@ def _getFirstFileByHash(self, algo, hash, user=None):
         return None
 
 
+def _computeHashHook(event):
+    """
+    Computes all supported checksums on a given file. Downloads the
+    file data and stream-computes all required hashes on it, saving
+    the results in the file document.
+    """
+    if ModelImporter.model('setting').get(PluginSettings.AUTO_COMPUTE, default=False):
+        _computeHash(event.info['file'])
+
+
+def _computeHash(file, progress=noProgress):
+    toCompute = SUPPORTED_ALGORITHMS - set(file)
+    toCompute = {alg: getattr(hashlib, alg)() for alg in toCompute}
+
+    if not toCompute:
+        return
+
+    fileModel = ModelImporter.model('file')
+    with fileModel.open(file) as fh:
+        while True:
+            chunk = fh.read(65536)
```

Comment:
> Maybe we should centralize this value, as we have it in a few places. Is it well determined, or semi-arbitrary, both?

semi-arbitrary. I haven't profiled running different-sized chunks through hash algs; I assume we'd be waiting on the network socket anyway, so probably not a huge deal.